src/passes/inlineAssemblyTransformer.ts

```diff
@@ -0,0 +1,106 @@
+import { ASTMapper } from '../ast/mapper';
+import { AST } from '../ast/ast';
+import { createExpressionStatement, createIdentifier, createTuple } from '../utils/nodeTemplates';
+
+import {
+  InlineAssembly,
+  YulNode,
+  Identifier,
+  ASTNode,
+  BinaryOperation,
+  Assignment,
+  Block,
+  Expression,
+  ExpressionStatement,
+  VariableDeclaration,
+  UncheckedBlock,
+} from 'solc-typed-ast';
+import assert from 'assert';
+
+class YulTransformer {
+  vars: { [name: string]: VariableDeclaration };
+  ast: AST;
+
+  constructor(ast: AST, externalReferences: any[], contextMap: Map<number, ASTNode>) {
+    this.ast = ast;
+    const blockRefIds = externalReferences.map((ref) => ref.declaration);
+    const blockRefs: [number, ASTNode][] = [...contextMap].filter(([id, _]) =>
+      blockRefIds.includes(id),
+    );
+    this.vars = Object.fromEntries(
+      blockRefs.map(([id, ref]) => [(ref as Identifier).name, ref as VariableDeclaration]),
+    );
+  }
+
+  run(node: YulNode): ASTNode {
+    if (!Object.hasOwn(this, node.nodeType)) throw `${node.nodeType} is not supported`;
+    const method = this[node.nodeType as keyof YulTransformer] as (node: YulNode) => ASTNode;
+    return method.bind(this)(node);
+  }
+
+  YulIdentifier(node: YulNode): Identifier {
```

Comment:
I would suggest renaming this funcs: `YulIdentifier`, `YulAssignment etc... to: `createYulIdentifier`, `createYulAssignment`